ci/ray_ci/automation/query_flaky_macos_test.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python3
+from ray_release.test import Test, TestState
+from ci.ray_ci.utils import logger
+from typing import List
+from ray_release.configs.global_config import init_global_config
+from ray_release.bazel import bazel_runfile
+import json
+
+ALL_TEST_PREFIXES = ["darwin:"]
+
+def main() -> None:
+    logger.info("Start querying flaky tests ...")
+    init_global_config(bazel_runfile("release/ray_release/configs/oss_config.yaml"))
+    tests = query_all_tests()
+    flaky_tests = filter_flaky_tests(tests)
+    logger.info("Number of flaky tests found: " + str(len(flaky_tests)))
+    logger.info("List of flaky tests: ")
+    print(retrieve_test_names(flaky_tests, 'darwin://'))
+
+def query_all_tests() -> List[Test]:
```

Comment:
move these functions into `ray_release.test` maybe ? or a `test_utils` file somewhere, make it a library, and add unit tests?